internal/fsapi/file.go

```diff
@@ -380,3 +375,158 @@ type File interface {
+	//   - However implementations will generally reopen the underlying directory
+	//     for portability reasons.
+	Reset() syscall.Errno
+
+	// Rewind seeks the internal cursor to the given offset.
+	//
+	// Rewinding to offset 0 usually should produce the same result
+	// as calling Reset.
+	//
+	// Implementations may keep a buffer or a moving window. The size of
+	// the window is an implementation detail, the implementation
+	// may return an error if the offset is too small to seek back
+	// to that index (i.e. it lays outside the window).
+	//
+	// # Errors
+	//
+	// A zero syscall.Errno is success. The below are expected otherwise:
+	//   - syscall.EINVAL: the offset is larger than the current internal cursor.
+	//   - syscall.ENOSYS: the offset is too small for the current window.
+	//
+	// # Notes
+	//
+	//   - This is conceptually similar to invoking Seek(offset, io.SeekStart)
+	//     on io.Seeker and `fseek` in POSIX over an open directory. See
+	//     https://pubs.opengroup.org/onlinepubs/9699919799/functions/fseek.html
+	//   - However implementations will generally keep a buffer of the currently
+	//     valid values and will not actually seek to the given offset the underlying
+	//     open directory.
+	Rewind(offset uint64) syscall.Errno
+
+	// Peek emits the value currently pointed by the internal cursor.
+	//
+	// # Errors
+	//
+	// A zero syscall.Errno is success. The below are expected otherwise:
+	//   - syscall.ENOENT: there are no more entries to fetch
+	//   - syscall.EBADF: the directory is no longer valid
+	//   - other error values would signal an issue with fetching the next batch of values.
+	//
+	// # Notes
+	//
+	//   - Ultimately, implementations may fetch data using a `readdir`-like API, thus
+	//     errors will reflect file system errors similarly to the POSIX call. See
+	//     https://pubs.opengroup.org/onlinepubs/9699919799/functions/readdir.html
+	Peek() (*Dirent, syscall.Errno)
+
+	// Next advances the internal cursor to the next value.
+	//
+	// This is equivalent to invoking Peek() and then advancing the internal
+	// cursor; thus, the error conditions are the same.
+	//
+	// # Errors
+	//
+	// A zero syscall.Errno is success. The below are expected otherwise:
+	//   - syscall.ENOENT: there are no more entries to fetch
+	//   - syscall.EBADF: the directory is no longer valid
+	//   - other error values would signal an issue with fetching the next batch of values.
+	//
+	// # Notes
+	//
+	//   - Ultimately, implementations may fetch data using a `readdir`-like API, thus
+	//     errors will reflect file system errors similarly to the POSIX call. See
+	//     https://pubs.opengroup.org/onlinepubs/9699919799/functions/readdir.html
+	Next() (*Dirent, syscall.Errno)
+
+	// Close closes the underlying file.
+	//
+	// # Errors
+	//
+	// A zero syscall.Errno is success. The below are expected otherwise:
+	//   - syscall.ENOSYS: the implementation does not support this function.
```

Comment:
Sorry our docs on fsapi.File Close are wrong. We shouldn't (and don't) return an error on not implemented close. Can you correct that, also?

`A zero syscall.Errno is returned if unimplemented or success.`

Reply:
TBH this was wrong too, it actually returns 0 also here.